src/common/snippets/src/lowered/pass/split_loops.cpp

```diff
@@ -54,18 +54,24 @@ bool SplitLoops::run(LinearIR& linear_ir, lowered::LinearIR::constExprIt begin,
                 continue;
 
             const auto& parent_loop_id = parent_loop_ids.front();
-            const auto parent_loop_port = loop_manager->get_loop_port_by_expr_port(parent_port, parent_loop_id);
+            const auto parent_loop = loop_manager->get_loop_info(parent_loop_id);
+
+            const bool split_parent = parent_loop->get_increment() < loop->get_increment();
+            const auto upper_loop = std::make_shared<LoopManager::LoopInfo>(*parent_loop);
+            const auto lower_loop = std::make_shared<LoopManager::LoopInfo>(*loop);
+            if (split_parent)
+                upper_loop->set_increment(loop->get_increment());
+            else
+                lower_loop->set_increment(parent_loop->get_increment());
             // We don't split loop which are not compatible with parent loop because such loops will not be fused
-            if (!FuseLoops::loop_ports_are_compatible(loop_manager, loop_id, parent_loop_id))
+            if (!FuseLoops::can_be_fused(upper_loop, lower_loop))
```

Comment:
Minor: we can combine 2 checks here now:
```suggestion
            if (!FuseLoops::can_be_fused(upper_loop, lower_loop) || !can_be_split(loop, parent_loop))
```

Reply:
Agree, done, thanks!